pkg/clients/client.go

```diff
@@ -53,11 +51,14 @@ func restConfigFromAPIConfig(c *api.Config) (*rest.Config, error) {
 	ctx := c.Contexts[c.CurrentContext]
 	cluster := c.Clusters[ctx.Cluster]
 	if cluster == nil {
-		return nil, errors.New(fmt.Sprintf("cluster for currentContext (%s) not found", c.CurrentContext))
+		return nil, errors.Errorf("cluster for currentContext (%s) not found", c.CurrentContext)
 	}
 	user := c.AuthInfos[ctx.AuthInfo]
 	if user == nil {
-		return nil, errors.New(fmt.Sprintf("auth info for currentContext (%s) not found", c.CurrentContext))
+		// We don't require a user because it's possible user
+		// authorization configuration will be loaded from a separate
+		// set of identity credentials (e.g. Google Application Creds).
+		user = &api.AuthInfo{}
```

Comment:
This change might not actually be necessary. I interpreted https://github.com/crossplane/provider-gcp/issues/343 as saying that there would be no entry in the `users` list in some cases, but for our GKE cluster implementation it's more that the user has an empty `user` block - i.e.:

```yaml
apiVersion: v1
clusters:
- cluster:
    certificate-authority-data: REDACTED
    server: REDACTED
  name: scale-test-0-c2rvh
contexts:
- context:
    cluster: scale-test-0-c2rvh
    user: scale-test-0-c2rvh
  name: scale-test-0-c2rvh
current-context: scale-test-0-c2rvh
kind: Config
preferences: {}
users:
- name: scale-test-0-c2rvh
  user: {}
```

Reply:
I'm going to leave this out of laziness since it can't hurt. 🤷